playwright-e2e/dsm/pages/participant-list-page.ts

```diff
@@ -247,20 +248,19 @@ export default class ParticipantListPage {
       const [isAddressValid] = await participantListTable.getTextAt(index, validColumn);
       const [country] = await participantListTable.getTextAt(index, countryColumn);
       const [state] = await participantListTable.getTextAt(index, stateOfResidence);
-      console.log(`Allow NY or Canadians: ${allowNewYorkerOrCanadian}`);
-      console.log(`PT info - country: ${country}, state: ${state}, valid: ${isAddressValid}`);
-      let isMatch = true;
+      logInfo(`Allow NY or Canadians: ${allowNewYorkerOrCanadian}`);
+      logInfo(`PT info - country: ${country}, state: ${state}, valid: ${isAddressValid}`);
+
+      let matchFirstName = true;
       if (firstNameSubstring) {
-        isMatch = fname.indexOf(firstNameSubstring) !== -1
-      }
-      if (isMatch && normalCollaboratorSampleID.length <= 5 && isAddressValid.toLowerCase() === 'true') {
-        return index;
+        matchFirstName = fname.indexOf(firstNameSubstring) !== -1;
       }
-      if (!allowNewYorkerOrCanadian &&
-        ((country === 'CA') ||
-        ((country === 'US') && (state === 'NY')))) {
+      if (!allowNewYorkerOrCanadian && (country === 'CA' || (country === 'US' && state === 'NY'))) {
         return -1;
       }
+      if (matchFirstName && normalCollaboratorSampleID.length <= 5 && isAddressValid.toLowerCase().trim() === 'true') {
+        return index;
+      }
```

Comment:
@Rinchan20  Two if-condition checks were in the wrong order. It led to "bad" (unsuitable) participants.

```
if (!allowNewYorkerOrCanadian && (country === 'CA' || (country === 'US' && state === 'NY')))
```
and 
```
if (matchFirstName && normalCollaboratorSampleID.length <= 5 && isAddressValid.toLowerCase().trim() === 'true')
```